test/space/ntuples_test.py

```diff
@@ -1609,9 +1609,43 @@ def other_dist(x, y):
         FnCustomDist(1)
 
 
-def _impl_test_ufuncs(fn, name, n_args, n_out):
+def test_ufuncs(fn, ufunc):
+    name, n_args, n_out, _ = ufunc
+    if (np.issubsctype(fn.dtype, np.floating) or
+            np.issubsctype(fn.dtype, np.complexfloating)and
+            name in ['bitwise_and',
+                     'bitwise_or',
+                     'bitwise_xor',
+                     'invert',
+                     'left_shift',
+                     'right_shift']):
+        # Skip integer only methods if floating point type
+        return
+
+    if (np.issubsctype(fn.dtype, np.complexfloating) and
+            name in ['remainder',
+                     'trunc',
+                     'signbit',
+                     'invert',
+                     'left_shift',
+                     'right_shift',
+                     'rad2deg',
+                     'deg2rad',
+                     'copysign',
+                     'mod',
+                     'modf',
+                     'fmod',
+                     'logaddexp2',
+                     'logaddexp',
+                     'hypot',
+                     'arctan2',
+                     'floor',
+                     'ceil']):
+        # Skip real only methods for complex
```

Comment:
perhaps not a good formulation but i want to say: "skip methods that only work on real numbers if dtype is complex". Will update that.